ui/ducks/send/send.js

```diff
@@ -1570,6 +1635,34 @@ export function signTransaction() {
           to: undefined,
           data: undefined,
         });
+        dispatch(showConfTxPage());
+        dispatch(hideLoadingIndication());
+      } catch (error) {
+        dispatch(hideLoadingIndication());
+        dispatch(displayWarning(error.message));
+      }
+    } else if (asset.type === ASSET_TYPES.COLLECTIBLE) {
+      // When sending a collectible transaction we have to use the collectible.transferFrom method
+      // on the collectible contract to construct the transaction. This results in
+      // the proper transaction data and properties being set and a new
+      // transaction being added to background state. Once the new transaction
+      // is added to state a subsequent confirmation will be queued.
+      try {
+        const collectibleContract = global.eth
+          .contract(abiERC721)
+          .at(asset.details.address);
+
+        collectibleContract.transferFrom(
+          selectedAddress,
+          address,
+          asset?.details?.tokenId,
```

Comment:
It looks like this can be null. Is that allowed for ERC721? If not, is it worth checking for this and throwing an explicit error if this is not set correctly?

Reply:
Actually sorry this optional chaining isn't necessary on lines 522-528:
```
if (asset.type === ASSET_TYPES.COLLECTIBLE) {
      if (asset.details === null) {
        // If we're sending a collectible but details have not been provided we must
        // abort and set the send slice into invalid status.
        throw new Error(
          'Send slice initialized as collectibles send without token details',
        );
      }
  ```
  We throw an error if the details aren't present.

Reply:
Okay cool. Just checking!

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/13048/commits/e0a1b4c80da158cc529475d6cdf7faca39d269d6